apps/mark-scan/frontend/src/pages/poll_worker_screen.tsx

```diff
@@ -181,9 +178,14 @@ export function PollWorkerScreen({
   const getStateMachineStateQuery = getStateMachineState.useQuery();
   const stateMachineState = getStateMachineStateQuery.data;
 
+  const getPrecinctSelectionQuery = getPrecinctSelection.useQuery();
+  const precinctSelection = getPrecinctSelectionQuery.data?.ok();
```

Comment:
I think this is actually masking a subtle bug. If `getPrecinctSelectionQuery` is being loaded for the first time, on the initial render, its `data` will be `undefined`. That will cause the default value for `selectedCardlessVoterPrecinctId` to be `undefined`. Once the query loads and rerenders, `selectedCardlessVoterPrecinctId` won't be updated, since the default value for the `useState` hook only applies on the first render. This may not happen in practice if the query data is already cached before this component is rendered, but it's still a latent bug waiting to happen.

I recommend always checking `query.isSuccess` before accessing `data` because it's easy for things to behave in weird ways otherwise. In this case, I would either pass in the precinct selection from a parent component or make a wrapper parent component to load the necessary data, putting the `useState` in the child component. It's a bit verbose but I think it's important.

Reply:
Thanks for flagging this bug. https://github.com/votingworks/vxsuite/pull/3786 returns `PollWorkerScreen` to its original contract of requiring `precinctSelection` to be passed as a non-optional prop.